tests/test_chebyMap.py

```diff
@@ -0,0 +1,493 @@
+    """
+    if n == 0:
+        return 1.0
+    if n == 1:
+        return x
+    return (2.0 * x * cheby(n-1, x)) - cheby(n-2, x)
+
+
+def normalize(inArray, lbnd, ubnd):
+    """Return the value of x normalized to [-1, 1]
+
+    Parameters
+    ----------
+    inArray : `numpy.array`
+        Value(s) to normalize; a list of nPoints x nAxes values
+        (the form used by astshim.Mapping.tranForward)
+    lbnd : sequence of `float`
+        Lower bounds (one element per axis)
+    ubnd : sequence of `float`
+        Upper bounds (one element per axis)
+    """
+    # normalize x in the range [-1, 1]
+    lbnd = np.array(lbnd)
+    ubnd = np.array(ubnd)
+    delta = ubnd - lbnd
+    return -1 + ((inArray - lbnd) * 2.0 / delta)
+
+
+class ReferenceCheby(object):
+
+    def __init__(self, referenceCheby, lbnd, ubnd):
+        """Construct a reference Chebyshev polynomial
+
+        Parameters
+        ----------
+        referenceCheby : callable
+            A function that takes a normalized point (as a list of floats)
+            that has been normalized to the range [-1, 1]
+            and returns the expected results from ChebyPoly.tranForward
+            or tranInverse for the corresponding un-normalized point
+        lbnd : list of float
+            Lower bounds of inputs (for normalization)
+        ubnd : list of float
+            Upper bounds of inputs (for normalization)
+        """
+        self.referenceCheby = referenceCheby
+        self.lbnd = lbnd
+        self.ubnd = ubnd
+
+    def transform(self, inArray):
+        """Transform data using the reference function
+
+        Parameters
+        ----------
+        inArray : `numpy.array`
+            Input array of points in the form used by ChebyMap.tranForward
+            or tranInverse.
+
+        Returns
+        -------
+        outArray : `numpy.array`
+            inArray transformed by referenceCheby (after normalizing inArray)
+        """
+        inNormalized = normalize(inArray, self.lbnd, self.ubnd)
+        outPoints = [self.referenceCheby(inPoint) for inPoint in inNormalized]
+        arr = np.array(outPoints)
+        if len(arr.shape) > 2:
+            # trim unwanted extra dimension (occurs when nin=1)
+            arr.shape = arr.shape[0:2]
+        return arr
+
+
+class TestChebyMap(MappingTestCase):
+
+    def setUp(self):
+        self.normErr = "Invalid {0} normalization: min={1}, max={2}, min/max norm=({3}, {4}) != (-1, 1)"
+        # We need a slightly larger than the full floating point tolerance for
+        # many of these tests.
+        self.atol = 5e-14
+
+    def test_chebyMapUnidirectional_2_2(self):
+        """Test one-directional ChebyMap with 2 inputs and 2 outputs
+
+        This is a long test because it is a bit of a nuisance setting up
+        the reference transform, so once I have it, I use it for three
+        different ChebyMaps (forward-only, forward with no inverse,
+        and inverse with no forward).
```

Comment:
I'm personally happy with it where it is. It's not as if this is part of a public API of a reusable object.